src/main/java/ua/tc/marketplace/jwtAuth/JwtUtil.java

```diff
@@ -0,0 +1,81 @@
+package ua.tc.marketplace.jwtAuth;
+
+import io.jsonwebtoken.*;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+import ua.tc.marketplace.model.entity.User;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@AllArgsConstructor
+public class JwtUtil {
+    private final JwtConfig jwtConfig;
+
+    public String createToken(User user) {
+        Claims claims = Jwts.claims().setSubject(user.getEmail());
+        claims.put("firstName",user.getFirstName());
+        claims.put("lastName",user.getLastName());
+        Date tokenCreateTime = new Date();
+        Date tokenValidity = new Date(tokenCreateTime.getTime()
+                + TimeUnit.MINUTES.toMillis(jwtConfig.getTokenExpirationAfterDays()*24*60));
+        return Jwts.builder()
+                .setClaims(claims)
+                .setExpiration(tokenValidity)
+                .signWith(SignatureAlgorithm.HS256, jwtConfig.getSecretKey())
+                .compact();
+    }
+
+    private Claims parseJwtClaims(String token) {
+        return Jwts.parser()
+                .setSigningKey(jwtConfig.getSecretKey())
+                .parseClaimsJws(token).getBody();
+    }
+
+    public Claims resolveClaims(HttpServletRequest req) {
+        try {
+            String token = resolveToken(req);
+            if (token != null) {
+                return parseJwtClaims(token);
+            }
+            return null;
+        } catch (ExpiredJwtException ex) {
+            req.setAttribute("expired", ex.getMessage());
+            throw ex;
+        } catch (Exception ex) {
+            req.setAttribute("invalid", ex.getMessage());
+            throw ex;
+        }
+    }
+
+    public String resolveToken(HttpServletRequest request) {
+
+        String bearerToken = request.getHeader(jwtConfig.getAuthorizationHeader());
+        if (bearerToken != null && bearerToken.startsWith(jwtConfig.getTokenPrefix())) {
+            return bearerToken.substring(jwtConfig.getTokenPrefix().length());
+        }
+        return null;
+    }
+
+    public boolean validateClaims(Claims claims) throws AuthenticationException {
+        try {
+            return claims.getExpiration().after(new Date());
+        } catch (Exception e) {
+            throw e;
```

Comment:
it is incorrect to catch e and throw e. Try to understand what happens here and throw appropriate custom exception if needed

Reply:
moved exception to calling method for now. But I still dont like it, will again search for better way to catch expiration exception